Dwarf Arcane Journal Updates

EmilyChilly opened the pull request:
Updated the Dwarf, Empire (mercenaries), and Magic items data. First time contributing so hopefully everything looks okay! (Made no changes to runic item points limits as discussed on the discord)

nthiebes commented on public/games/the-old-world/empire-of-man.json:
```diff
@@ -2890,6 +2890,185 @@
     }
   ],
-  "mercenaries": [],
+  "mercenaries": [
```
Mercenary data always lives inside the dataset of the owning faction to avoid duplication.

nthiebes reviewed (commented):
Thanks for the effort, that's very helpful! Your changes look good, I only had one comment without checking all the details of each unit.

One additional request: Could you choose https://github.com/nthiebes/old-world-builder/tree/global-errors as base branch? 🙏 That's the one I'm currently working in, which will include the arcane journal armies and some global list validation.

EmilyChilly commented:
Glad to hear I did alright! I've updated the base branch as requested and removed the changes to the empire mercenaries. Quesiton: Is there somewhere else I would flag which factions can take said mercenaries, or is that handled deeper in the app somewhere?

nthiebes commented:
They're configured here: https://github.com/nthiebes/old-world-builder/blob/main/src/assets/armies.json#L18

EmilyChilly commented:
Thanks! I went ahead and updated the mercenary lists as well
